From bcc05d3d0de7fc842afbc355e69c531267628de2 Mon Sep 17 00:00:00 2001
From: Andrzej Ostruszka <amo@semihalf.com>
Date: Mon, 14 Jun 2021 16:03:02 +0000
Subject: [PATCH 18/19] Send more DHCPv6 options over DBus

This is the same patch as the one for 6.8.2 version, but dressed as
a formal git patch.  Doing this makes it easy to take all patches from
overlay, concatenate them in right order and apply them via git-am.
---
 src/dbus/rpc-dbus.c | 43 ++++++++++++++++++++++++++++++++++++++-----
 1 file changed, 38 insertions(+), 5 deletions(-)

diff --git a/src/dbus/rpc-dbus.c b/src/dbus/rpc-dbus.c
index 22608d15..9f3ae512 100644
--- a/src/dbus/rpc-dbus.c
+++ b/src/dbus/rpc-dbus.c
@@ -224,19 +224,52 @@ static const struct o_dbus dhos[] = {
 #ifdef INET6
 	{ "dhcp6_server_id=", DBUS_TYPE_STRING, 0,
 	  "DHCPv6ServerIdentifier" },
-	{ "dhcp6_ia_na1_ia_addr1=", DBUS_TYPE_STRING, 0, "DHCPv6Address" },
+	{ "dhcp6_ia_na1_iaid=", DBUS_TYPE_UINT32, 0, "DHCPv6AddressIAID" },
+	{ "dhcp6_ia_na1_ia_addr1=", DBUS_TYPE_STRING, 0, "DHCPv6Address1" },
 	{ "dhcp6_ia_na1_ia_addr1_vltime=", DBUS_TYPE_UINT32, 0,
-	  "DHCPv6AddressLeaseTime" },
+	  "DHCPv6AddressLeaseTime1" },
+	{ "dhcp6_ia_na1_ia_addr1_pltime=", DBUS_TYPE_UINT32, 0,
+	  "DHCPv6AddressPreferredLeaseTime1" },
+	{ "dhcp6_ia_na1_ia_addr2=", DBUS_TYPE_STRING, 0, "DHCPv6Address2" },
+	{ "dhcp6_ia_na1_ia_addr2_vltime=", DBUS_TYPE_UINT32, 0,
+	  "DHCPv6AddressLeaseTime2" },
+	{ "dhcp6_ia_na1_ia_addr2_pltime=", DBUS_TYPE_UINT32, 0,
+	  "DHCPv6AddressPreferredLeaseTime2" },
+	{ "dhcp6_ia_na1_ia_addr3=", DBUS_TYPE_STRING, 0, "DHCPv6Address3" },
+	{ "dhcp6_ia_na1_ia_addr3_vltime=", DBUS_TYPE_UINT32, 0,
+	  "DHCPv6AddressLeaseTime3" },
+	{ "dhcp6_ia_na1_ia_addr3_pltime=", DBUS_TYPE_UINT32, 0,
+	  "DHCPv6AddressPreferredLeaseTime3" },
 	{ "dhcp6_name_servers=", DBUS_TYPE_ARRAY, DBUS_TYPE_STRING,
 	  "DHCPv6NameServers" },
 	{ "dhcp6_domain_search=", DBUS_TYPE_ARRAY, DBUS_TYPE_STRING,
 	  "DHCPv6DomainSearch" },
+	{ "dhcp6_ia_pd1_iaid=", DBUS_TYPE_UINT32, 0,
+	  "DHCPv6DelegatedPrefixIAID" },
 	{ "dhcp6_ia_pd1_prefix1=", DBUS_TYPE_STRING, 0,
-	  "DHCPv6DelegatedPrefix" },
+	  "DHCPv6DelegatedPrefix1" },
 	{ "dhcp6_ia_pd1_prefix1_length=", DBUS_TYPE_UINT32, 0,
-	  "DHCPv6DelegatedPrefixLength" },
+	  "DHCPv6DelegatedPrefixLength1" },
 	{ "dhcp6_ia_pd1_prefix1_vltime=", DBUS_TYPE_UINT32, 0,
-	  "DHCPv6DelegatedPrefixLeaseTime" },
+	  "DHCPv6DelegatedPrefixLeaseTime1" },
+	{ "dhcp6_ia_pd1_prefix1_pltime=", DBUS_TYPE_UINT32, 0,
+	  "DHCPv6DelegatedPrefixPreferredLeaseTime1" },
+	{ "dhcp6_ia_pd1_prefix2=", DBUS_TYPE_STRING, 0,
+	  "DHCPv6DelegatedPrefix2" },
+	{ "dhcp6_ia_pd1_prefix2_length=", DBUS_TYPE_UINT32, 0,
+	  "DHCPv6DelegatedPrefixLength2" },
+	{ "dhcp6_ia_pd1_prefix2_vltime=", DBUS_TYPE_UINT32, 0,
+	  "DHCPv6DelegatedPrefixLeaseTime2" },
+	{ "dhcp6_ia_pd1_prefix2_pltime=", DBUS_TYPE_UINT32, 0,
+	  "DHCPv6DelegatedPrefixPreferredLeaseTime2" },
+	{ "dhcp6_ia_pd1_prefix3=", DBUS_TYPE_STRING, 0,
+	  "DHCPv6DelegatedPrefix3" },
+	{ "dhcp6_ia_pd1_prefix3_length=", DBUS_TYPE_UINT32, 0,
+	  "DHCPv6DelegatedPrefixLength3" },
+	{ "dhcp6_ia_pd1_prefix3_vltime=", DBUS_TYPE_UINT32, 0,
+	  "DHCPv6DelegatedPrefixLeaseTime3" },
+	{ "dhcp6_ia_pd1_prefix3_pltime=", DBUS_TYPE_UINT32, 0,
+	  "DHCPv6DelegatedPrefixPreferredLeaseTime3" },
 #endif
 	{ NULL, 0, 0, NULL }
 };
-- 
2.33.0.800.g4c38ced690-goog

